lib/lte_link_control/lte_lc.c

```diff
@@ -174,11 +174,12 @@ static const char *const system_mode_params[] = {
 
 static struct k_sem link;
 
-#if defined(CONFIG_LTE_PDP_CMD) && defined(CONFIG_LTE_PDP_CONTEXT)
-static const char cgdcont[] = "AT+CGDCONT="CONFIG_LTE_PDP_CONTEXT;
+#if defined(CONFIG_LTE_PDP_CMD)
+static char cgdcont[144] = "AT+CGDCONT="CONFIG_LTE_PDP_CONTEXT;
 #endif
-#if defined(CONFIG_LTE_PDN_AUTH_CMD) && defined(CONFIG_LTE_PDN_AUTH)
-static const char cgauth[] = "AT+CGAUTH="CONFIG_LTE_PDN_AUTH;
+#if defined(CONFIG_LTE_PDN_AUTH_CMD)
+static char cgauth[19 + CONFIG_LTE_PDN_AUTH_LEN] =
```

Comment:
Why `19 +`? Anyway, I think it would be simpler to specify maximum command length via Kconfig, not only part of it.

Reply:
I imagine the user would set `CONFIG_LTE_PDN_AUTH_LEN` to length of `CONFIG_LTE_PDN_AUTH_CMD` and would not know to account (or could account) for the prefix that we add to it before sending it to the modem. That's why I am adding it manually here. Otherwise we'd have to say something in Kconfig like add 19 to the length of your string which is clunky IMHO.